### PR TITLE
Document solver specific options

### PR DESCRIPTION
**Pull request by delcypher**
This documents existing solver specific options and the new Z3 specific options in https://github.com/klee/klee/pull/658

### CODE REVIEW

**Comment by delcypher** [comment]
Options have now been rewritten with the `-debug-` prefix.

File: docs/solver-specific-options.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+### `-solver-optimize-divides`
+
+This only affects the MetaSMT and STP backends. When set to true expressions involving
+division will be optimized (if possible) before being given to the solver backend.
+
+### `-use-forked-solver`
+
+This only affects the MetaSMT and STP backends. When set to true (**default:
+true**)the solver is created in a seperate process that is forked from the
```
Comment (andreamattavelli) [inline docs/solver-specific-options.md]: Missing space after bracket )

File: docs/solver-specific-options.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+### `-use-construct-hash`
+
+When set to true (**default: true**) constructed STP expressions will be cached
+(and then cleared) for each constraint to facilitate expression re-use.
+
+# MetaSMT
+
+These are options that only affect the MetaSMT backend.
+
+### `use-construct-hash-metasmt`
```
Comment (andreamattavelli) [inline docs/solver-specific-options.md]: Typo: should be`-use-construct-hash-metasmt`

**Comment by delcypher** [comment]
@andreamattavelli Thanks for spotting the typos. I've fixed them.

**Comment by delcypher** [comment]
@andreamattavelli @ccadar Can we merge this?

**Comment by andreamattavelli** [comment]
@delcypher As I previously mentioned, I am not extremely happy about dividing `Solver chain` and `Solver specific options`. @ccadar Any thoughts about this?

**Comment by delcypher** [comment]
@andreamattavelli Oops. I missed your previous coment about merging "solver chain" and "solver specific options". I have no objections to this. Seems like a good idea because it will make the options easier to find.

**Comment by andreamattavelli** [comment]
@delcypher 
> Seems like a good idea because it will make the options easier to find.

Exactly!

**Comment by delcypher** [comment]
@andreamattavelli Okay I've merged the documentation into the existing documentation about the solver chain. Please take a look.

Personally I don't like our inconsistent use of "core solver" and "solver backend" in KLEE. I'd rather we had just one term for this. I prefer "solver backend".

**Pull request merged**